nautobot/core/celery/backends.py

```diff
@@ -1,7 +1,60 @@
 from django_celery_results.backends import DatabaseBackend
 
+from nautobot.extras.models import JobResult
+
 
 class NautobotDatabaseBackend(DatabaseBackend):
     """
     Nautobot extensions to support database integration of Job machinery.
     """
+
+    TaskModel = JobResult
+
+    def encode_content(self, data):
+        """Pass through encoding since we're storing as JSON explicitly."""
+        return "application/x-nautobot-json", "utf-8", data
+
+    def decode_content(self, obj, content):
+        """Pass through decoding since we're storing as JSON explicitly."""
+        return content
+
+    def _get_extended_properties(self, request, traceback):
+        """
+        Overload default so that `argsrepr` and `kwargsrepr` aren't used to construct `args` and
+        `kwargs`. Our need to properly (de)serialize `NautobotFakeRequest` facilitates this.
```

Comment:
If we get rid of `NautobotFakeRequest`, will that get rid of the need to override this private method? Not saying that has to be done as part of this PR though.

Reply:
Yes, I think so. The other problem are the nullable `CharField` fields we forklifted. If we want "correctness" I would rather those default to `""` vs. `None` but that's actually more invasive than what I had to do here.